### PR TITLE
Fix NUMERIC support (map to Data.Scientific)

### LINKED ISSUE #1017: Getting Incompatible error when using Numeric column
```
Incompatible {errSQLType = "numeric", errSQLTableOid = Just (Oid 17158), errSQLField = "cost",
 errHaskellType = "Float", errMessage = "types incompatible"}
```

Apparently the generated model type is Float, which is obviously incorrect. "Numeric" is a fixed-point number, so I would expect it to map to `Data.Fixed.Fixed`. As it is, it just doesn't work, apparently.

### PR DESCRIPTION
**Pull request by lierdakil**
Fixes #1017 

I've added no tests, but there were no tests for this to begin with.

### CODE REVIEW

File: IHP/Controller/Param.hs [hunk 1]
```diff
@@ -300,6 +300,16 @@ instance ParamReader Double where
                     Right integer -> Right (fromIntegral integer)
     readParameterJSON _ = Left "ParamReader Double: Expected Double"
 
+instance ParamReader Scientific.Scientific where
```
Comment (mpscholten) [inline IHP/Controller/Param.hs:303]: Tests for ParamReader instances can be added here https://github.com/digitallyinduced/ihp/blob/master/Test/Controller/ParamSpec.hs :) Could you add a case there? :)

File: IHP/ModelSupport.hs [hunk 2]
```diff
@@ -154,6 +155,9 @@ instance InputValue value => InputValue [value] where
 instance InputValue Aeson.Value where
     inputValue json = json |> Aeson.encode |> cs
 
+instance InputValue Scientific where
```
Comment (mpscholten) [inline IHP/ModelSupport.hs:158]: Tests for InputValue can be added here https://github.com/digitallyinduced/ihp/blob/master/Test/ModelSupportSpec.hs#L16 :)

**Comment by lierdakil** [comment]
I've added some tests (mostly copy-pasted from similar tests for `Double`).

File: Test/Controller/ParamSpec.hs [hunk 3]
```diff
@@ -141,32 +142,53 @@ tests = do
 
                 it "should accept floating point input" do
                     (readParameter @Double "1.2") `shouldBe` (Right 1.2)
-                    (readParameter @Float "1.2345679") `shouldBe` (Right 1.2345679)
-                
+                    (readParameter @Double "1.2345679") `shouldBe` (Right 1.2345679)
```
Comment (lierdakil) [inline Test/Controller/ParamSpec.hs:145]: While at it, I've fixed an apparent copy-paste error in `ParamReader Double` tests. Which reasonably speaking doesn't affect anything, but hey.

Comment (mpscholten) [reply Test/Controller/ParamSpec.hs:145]: Good catch :)

**Review by mpscholten — APPROVED** [review]
Thanks! :)

**Pull request merged**